rstar/src/algorithm/iterators.rs

```diff
@@ -84,11 +84,11 @@ where
     Func: SelectionFunction<T>,
 {
     pub fn new(root: &'a mut ParentNode<T>, func: Func) -> Self {
-        let current_nodes = root
-            .children
-            .iter_mut()
-            .filter(|c| func.should_unpack_parent(&c.envelope()))
```

Comment:
This assumed that `should_unpack_leaf` was just `self.should_unpack_parent(&leaf.envelope())`

Reply:
You're absolutely right. Thanks for finding and fixing this!